manifest/hash.go

```diff
@@ -0,0 +1,36 @@
+package manifest
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"errors"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/ovrclk/akash/types"
+)
+
+var ErrDifferentHashes = errors.New("manifest hash does not match the expected hash")
+
+func Hash(m proto.Message) ([]byte, error) {
+	bytes, err := marshal(m)
+	if err != nil {
+		return nil, err
+	}
+	h := sha256.New()
+	_, err = h.Write(bytes)
+	if err != nil {
+		return nil, err
+	}
+	return h.Sum(nil), err
+}
+
+func verifyHash(m *types.Manifest, expectedHash []byte) error {
```

Comment:
m can be github.com/ovrclk/akash/vendor/github.com/gogo/protobuf/proto.Message

Reply:
I'm gonna stop linting here and do another pr that covers the entire project

Reply:
I'm fine with just turning it off for now.  Just wanted to try it out.

Don't bother w/ fixing issues unless you think they're reasonable & it won't make the codebase too inconsistent.